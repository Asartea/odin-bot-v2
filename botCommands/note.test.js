const command = require('./note')

describe('/note', () => {
  describe('regex', () => {
    it.each([
      ['/note'],
      [' /note'],
      ['/note @odin-bot'],
      ['@odin-bot /note'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy()
    })
    
    it.each([
     ["ntoe"],
     ["note"],
     ["noet"],
     ["/noet"],
     ["/ntoe"],
     ["/ note"],
     ["/anote"]
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy()
    })

    it.each([
      ['Check this out! /note'],
      ['Don\'t worry about /note'],
      ['Hey @odin-bot, /note'],
      ['/@odin-bot ^ /me /note /tests$*']
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy()
    })

    it.each([
      ['@user/note'],
      ['it\'s about/note'],
      ['/notesanillusion'],
      ['/note/'],
      ['/note*'],
      ['/note...']
    ])("'%s' - command should be its own word/group - no leading or trailing characters", (string) => {
      expect(command.regex.test(string)).toBeFalsy()
    })
  })

  describe('callback', () => {
    it('returns correct output', () => {
      expect(command.cb()).toMatchSnapshot()
    })
  })
})