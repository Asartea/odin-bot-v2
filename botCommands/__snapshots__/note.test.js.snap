// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/note callback returns correct output 1`] = `"To note, or not to note, that is the question. Please read this: https://docs.google.com/document/d/1fas8xwMRi4GVrnCtp9U5OL7Y9QB6v5H_Wx19m4UQwgI/edit"`;
